autopush/base.py

```diff
@@ -57,16 +57,21 @@ def write_error(self, code, **kwargs):
         websocket.
 
         """
-        self.set_status(code)
-        if 'exc_info' in kwargs:
+        try:
+            self.set_status(code)
+            if 'exc_info' in kwargs:
+                self.log.failure(
+                    format=kwargs.get('format', "Exception"),
+                    failure=failure.Failure(*kwargs['exc_info']),
+                    client_info=self._client_info)
+            else:
+                self.log.failure("Error in handler: %s" % code,
+                                 client_info=self._client_info)
+            self.finish()
+        except Exception as ex:
             self.log.failure(
-                format=kwargs.get('format', "Exception"),
-                failure=failure.Failure(*kwargs['exc_info']),
-                client_info=self._client_info)
-        else:
-            self.log.failure("Error in handler: %s" % code,
-                             client_info=self._client_info)
-        self.finish()
+                "error in write_error: {} while printing {}".format(
```

Comment:
Can you include code in here too? (this "Uncaught exception in write_error: " error is really weird..)

Reply:
Will do. No idea if the exception is just not returning a value from `str(e)` or if there's some other weirdness going on, thus why I added the original data. And, yeah, it is.